052-cruise-control-api-users.md

```diff
@@ -0,0 +1,220 @@
+ADMIN role: has access to all endpoints.
+```
+
+The problem with this approach is that it would add significant complexity to the code base since it would require:
+* A new Strimzi `CruiseControlUser` custom resource
+* A watch on the new `CruiseControlUser` resource
+* A reconciler to manage the `CruiseControlUser` resource.
+ 
+Given that accessing the Cruise Control API directly is neither the supported or primary method of interacting with Strimzi Cruise Control, this method would not be worth the complexity that it would add to the code base.
+
+### Add API user list to Cruise Control spec 
+
+For adding Cruise Control API users, we could update the Cruise Control spec with an `apiUsers` section.
+In this `apiUsers` section, developers could provide a list of API users by name and security role.
+
+The schema could look something like this:
+
+```yaml=
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  ...
+spec:
+  cruiseControl:
+    apiUsers:
+      - name: user
+        role: USER (3)
+     ...
+```
+
+When defined like above, Strimzi would generate a Secret containing a password for the specified API user.
+
+In addition to the above, we could allow developers to supply a custom password per specified API user via a Secret like this:
+
+```yaml=
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  ...
+spec:
+  cruiseControl:
+    apiUsers:
+      - name: user
+        role: USER (3)
+        password:
+          valueFrom:
+            secretKeyRef:
+              name: my-secret (1)
+              key: my-password (2)
+     ...
+```
+
+(1) The name of the secret containing the predefined password.
+
+(2) The key for the password stored inside the secret.
+
+(3) By default Cruise Control defines three roles: `VIEWER`, `USER` and `ADMIN`. 
+For more details checkout the [Cruise Control Wiki](https://github.com/linkedin/cruise-control/wiki/Security#authorization)
+```
+VIEWER role: has access to the most lightweight kafka_cluster_state, user_tasks and review_board endpoints.
+USER role: has access to all the GET endpoints except bootstrap and train.
+ADMIN role: has access to all endpoints.
+```
+
+Strimzi would then aggregate the Cruise Control API auth credentials into a centralized configuration, then store it in a secret where it would be mounted and used by Cruise Control.
+
+However, given that accessing the Cruise Control API directly is neither the supported or primary method of interacting with Strimzi Cruise Control, this method would not be worth the complexity that it would add to the code base.
+
+## Future work
+
+This proposal aims to set the direction and define the semantics of Cruise Control API users and maybe set the precedent for creating API users for other Kafka components like KafkaConnect.
+It is not intended, by itself, to define the complete picture.
+Future work _could_ include:
+
+* Configurable API user roles
+* Password renewals for generated passwords
```

Comment:
Is this still relevant for the new method for managing the users? I would assume that the user just changes the secret to _renew_ passwords.